### PR TITLE
ENS support for sending tips and sending / receiving kudos

### PR DESCRIPTION
**Pull request by erib3**
##### Description

I added support for ENS by resolving addresses on the server side, it proved to be a simpler way to implement this. ENS resolution is now possible when sending tips and kudos and also receiving kudos. 

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/5470

### CODE REVIEW

File: app/assets/v2/js/shared.js [hunk 1]
```diff
@@ -977,6 +977,20 @@ function getNetwork(id) {
   return networks[id] || 'custom network';
 }
 
+function resolveENS(ens_name) {
+  let test_Url = 'http://localhost:8000/resolve_ens/';
+  let prod_Url = 'https://gitcoin.co/resolve_ens/';
+
+  var request = new XMLHttpRequest();
+
+  request.open('GET', test_Url.concat(ens_name), false); // `false` makes the request synchronous
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js:986]: code needs to be updated -> to use test in local / production

File: app/assets/v2/js/shared.js [hunk 2]
```diff
@@ -977,6 +977,20 @@ function getNetwork(id) {
   return networks[id] || 'custom network';
 }
 
+function resolveENS(ens_name) {
+  let test_Url = 'http://localhost:8000/resolve_ens/';
+  let prod_Url = 'https://gitcoin.co/resolve_ens/';
+
+  var request = new XMLHttpRequest();
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js:984]: ^ let /const 
Do we wanna use `fetchData` instead of `XMLHttpRequest`?

File: app/assets/v2/js/shared.js [hunk 3]
```diff
@@ -977,6 +977,20 @@ function getNetwork(id) {
...
+
+  request.open('GET', test_Url.concat(ens_name), false); // `false` makes the request synchronous
+  request.send(null);
+
+  if (request.status === 200) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js:989]: Could we also handle when status is not 200

**Comment by erib3** [comment]
https://drive.google.com/file/d/1In00O2YNDDG_8VUU3_u63t4RhOUDH2wH/view?usp=sharing

File: app/assets/v2/js/shared.js (hunk 2 above)
Comment (octavioamu) [reply app/assets/v2/js/shared.js:984]: for sure we should be using fetch instead.

**Comment by thelostone-mc** [comment]
Closing this out ! Do reopen this if comments can be addresed

**Pull request closed without merging**